src/commands/gen-js/library.ts

```diff
@@ -37,15 +42,10 @@ export function genJS(
       /**
        * Instantiate a wrapper around an analytics library instance
        * @param {Analytics} analytics - The ${clientName} library to wrap
-       * @param {Object} config - A configuration object to customize runtime behavior
        */
-      constructor(analytics, options = {}) {
-        const { propertyValidation = true } = options
-        if (!analytics) {
-          throw new Error('An instance of ${clientName} must be provided')
-        }
-        this.analytics = analytics
-        this.propertyValidation = propertyValidation
+      constructor(analytics) {
+        ${runtimeValidation ? analyticsValidation : ''}
+        this.analytics = analytics || { track: () => null }
```

Comment:
Shouldn't we always validate that an instance of a client is provided here?  Feel like this should be a "fail fast" situation

Reply:
CMIIW, but if this is used in a client-side application, `window.analytics` may be null if an ad blocker is enabled, which would cause Typewriter to crash when making analytics calls 😬 .

With this, it will still fail fast during development when you would have runtimeValidation on.